openstates/me/bills.py

```diff
@@ -249,37 +249,84 @@ def scrape_bill(self, bill, chamber):
                     vdoc.make_links_absolute(ver_url)
 
                     # various versions: billtexts, billdocs, billpdfs
-                    bill_text_html = vdoc.xpath(
-                        'string(//a[contains(@href, \
-                        "billtexts/")][1]/@href)'
-                    )
-                    bill_text_rtf = vdoc.xpath(
-                        'string(//a[contains(@href, \
-                        "billdocs/")][1]/@href)'
-                    )
-                    bill_text_pdf = vdoc.xpath(
-                        'string(//a[contains(@href, \
-                        "getPDF.asp")][1]/@href)'
-                    )
-
-                    if bill_text_html:
-                        bill.add_version_link(
-                            "Initial Version", bill_text_html, media_type="text/html"
+                    v_links = []
+                    for v in range(
+                        0, len(vdoc.xpath('//span[@class="story_heading"]')) - 1
+                    ):
+                        version_title = vdoc.xpath('//span[@class="story_heading"]')[
+                            v
+                        ].text
+                        version_pdf = vdoc.xpath(
+                            f"//span[@class='story_heading']/following::a[contains(@href, 'getPDF')]/@href"
                         )
-
-                    if bill_text_rtf:
-                        bill.add_version_link(
-                            "Initial Version",
-                            bill_text_rtf,
-                            media_type="application/rtf",
+                        version_html = vdoc.xpath(
+                            f"//span[@class='story_heading']/following::a[contains(@class, 'small_html_btn')]"
+                            "[contains(@href, 'asp')]/@href"
                         )
-
-                    if bill_text_pdf:
-                        bill.add_version_link(
-                            "Initial Version",
-                            bill_text_pdf,
-                            media_type="application/pdf",
+                        version_rtf = vdoc.xpath(
+                            f"//span[@class='story_heading']/following::a[contains(@href, 'rtf')]/@href"
+                        )
+                        version_fiscal_pdf = vdoc.xpath(
+                            f"//span[@class='story_heading']/following::a[contains(@href, 'fiscalpdfs')]/@href"
                         )
+                        version_fiscal_html = vdoc.xpath(
+                            f"//span[@class='story_heading']/following::a[contains(@href, 'fiscalnotes')]/@href"
+                        )
+
+                        if len(version_pdf) > v:
```

Comment:
I think this definitely needs some comments to be more clear, it is unclear what these checks are doing (what does it mean for the len(pdf) > v and why are the version_pdf[v] in v_links checks there?

Reply:
Added some additional comments to clear it up.